src/NuGet.Core/NuGet.Packaging/Signing/Authoring/SignPackageRequest.cs

```diff
@@ -80,19 +98,49 @@ public sealed class SignPackageRequest : IDisposable
         /// <param name="certificate">The signing certificate.</param>
         /// <param name="signatureHashAlgorithm">The signature hash algorithm.</param>
         /// <param name="timestampHashAlgorithm">The timestamp hash algorithm.</param>
+        /// <param name="signaturePlacement">The signature placement.</param>
+        /// <param name="v3ServiceIndexUrl">The V3 service index URL.</param>
+        /// <param name="packageOwners">A read-only list of package owners.</param>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="certificate" />
         /// is <c>null</c>.</exception>
         /// <exception cref="ArgumentException">Thrown if <paramref name="signatureHashAlgorithm" />
         /// is invalid.</exception>
         /// <exception cref="ArgumentException">Thrown if <paramref name="timestampHashAlgorithm" />
         /// is invalid.</exception>
-        /// <exception cref="ArgumentException">Thrown if <paramref name="signatureType" />
+        /// <exception cref="ArgumentException">Thrown if <paramref name="signaturePlacement" />
         /// is invalid.</exception>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="v3ServiceIndexUrl" />
+        /// is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="v3ServiceIndexUrl" />
+        /// is neither absolute nor HTTPS.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="packageOwners" />
+        /// is either empty or contains an invalid value.</exception>
         public SignPackageRequest(
             X509Certificate2 certificate,
             HashAlgorithmName signatureHashAlgorithm,
             HashAlgorithmName timestampHashAlgorithm,
-            SignatureType signatureType)
+            SignaturePlacement signaturePlacement,
+            Uri v3ServiceIndexUrl,
+            IReadOnlyList<string> packageOwners)
+            : this(
+                  certificate,
+                  signatureHashAlgorithm,
+                  timestampHashAlgorithm,
+                  SignatureType.Repository,
+                  signaturePlacement,
+                  v3ServiceIndexUrl,
+                  packageOwners)
+        {
+        }
+
+        private SignPackageRequest(
+            X509Certificate2 certificate,
+            HashAlgorithmName signatureHashAlgorithm,
+            HashAlgorithmName timestampHashAlgorithm,
+            SignatureType signatureType,
+            SignaturePlacement signaturePlacement,
+            Uri v3ServiceIndexUrl,
```

Comment:
wouldn't it be cleaner to have a custom object (e.g. `NuGetSignatureSignedAttributes`) that could have any attributes that are used either for author or repository signatures instead of having some variables that will be null is it's author and have data if it's repository?